docker-app/qfieldcloud/core/tests/test_username_generation.py

```diff
@@ -0,0 +1,47 @@
+import logging
+import random
+
+from django.http import HttpRequest
+from django.test.testcases import TransactionTestCase
+
+from qfieldcloud.core.adapters import AccountAdapter
+from qfieldcloud.core.models import Person
+from qfieldcloud.core.tests.utils import setup_subscription_plans
+
+logging.disable(logging.CRITICAL)
+
+
+class QfcTestCase(TransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+        self.existing_user = Person.objects.create_user(
+            username="existing9928", password="abc123"
+        )
+
+    def test_generated_usernames(self):
+        random.seed(42)
+        expectations = [
+            # Bases username on localpart of email plus random 4-digit suffix
+            ("john@example.org", "john2824"),
+            # Collisions with existing usernames are avoided
+            ("existing@example.org", "existing99286"),
```

Comment:
Would you mind splitting this test into a separate test case? Now it's a bit hidden behind. Also you can kill the `existing_user` from the `setUp` and use it only in the separate test case.